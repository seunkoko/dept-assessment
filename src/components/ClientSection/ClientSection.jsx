// eslint-disable-next-line
import React from 'react';

import './ClientSection.scss';

import abn from '../../assets/images/logos/ABN_logo_zwart.svg';
import adidas from '../../assets/images/logos/Adidas_logo_zwart.svg';
import klm from '../../assets/images/logos/KLM_logo_zwart.svg';
import microsoft from '../../assets/images/logos/Microsoft_logo_zwart-320x161.svg';
import mona from '../../assets/images/logos/Mona_logo_zwart.svg';
import nivea from '../../assets/images/logos/Nivea_logo_zwart.svg';
import nn from '../../assets/images/logos/NN_logo_zwart.svg';
import oxxio from '../../assets/images/logos/Oxxio_logo_zwart.svg';
import pathe from '../../assets/images/logos/Pathe_logo_zwart.svg';
import tomtom from '../../assets/images/logos/Tomtom_logo_zwart.svg';
import transavia from '../../assets/images/logos/Transavia_logo_zwart.svg';
import triumph from '../../assets/images/logos/Triumph_logo_zwart.svg';
import unilever from '../../assets/images/logos/Unilever_logo_zwart.svg';
import walibi from '../../assets/images/logos/Walibi_logo_zwart.svg';
import zalando from '../../assets/images/logos/Zalando_logo_zwart.svg';
import ziggo from '../../assets/images/logos/Ziggo_logo_zwart.svg';

const ClientSection = ({ id }) => {
	return (
    <>
      <div id={id} className="client-style">
        <p className="client-title">
          CLIENTS
        </p>
        <p className="client-description">
          We value a great working relationship with our clients 
          above all else. It’s why they often come to our parties. 
          It’s also why we’re able to challenge and inspire them to 
          reach for the stars.
        </p>

        <div className="client-logos">
          <div>
            <img src={nivea} alt="logo" />
          </div>
          <div>
            <img src={mona} alt="logo" />
          </div>
          <div>
            <img src={transavia} alt="logo" />
          </div>
          <div>
            <img src={zalando} alt="logo" />
          </div>

          <div>
            <img src={tomtom} alt="logo" />
          </div>
          <div>
            <img src={unilever} alt="logo" />
          </div>
          <div className="hide-on-phone">
            <img src={adidas} alt="logo" />
          </div>
          <div className="hide-on-phone">
            <img src={pathe} alt="logo" />
          </div>

          <div className="hide-on-phone">
            <img src={abn} alt="logo" />
          </div>
          <div className="hide-on-phone">
            <img src={triumph} alt="logo" />
          </div>
          <div className="hide-on-phone">
            <img src={microsoft} alt="logo" />
          </div>
          <div className="hide-on-phone">
            <img src={oxxio} alt="logo" />
          </div>

          <div className="hide-on-phone">
            <img src={nn} alt="logo" />
          </div>
          <div className="hide-on-phone">
            <img src={ziggo} alt="logo" />
          </div>
          <div className="hide-on-phone">
            <img src={walibi} alt="logo" />
          </div>
          <div className="hide-on-phone">
            <img src={klm} alt="logo" />
          </div>
        </div>
      </div>
    </>
	);
};

export default ClientSection;
