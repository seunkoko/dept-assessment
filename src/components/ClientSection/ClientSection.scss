@import "../../stylesheets/variables";

.client-style {
  position: absolute;
  left: 0;
  right: 20px;
  margin-bottom: 30px;
  padding: 50px;
  text-align: center;
  background-color: $client-color;
  letter-spacing: 1px;
  max-height: 700px;

  @media (max-width: $phone-width) {
    right: 0;
    padding: 20px;
	}
}

.client-title {
  font-size: $big-font;
  font-family: 'Teko-Regular';
}

.client-description {
  font-size: $medium-font;
  font-family: 'Teko-Light';
  margin: 0 auto 50px auto;
  width: 35%;

  @media (max-width: $laptop-width) {
		width: 50%;
	}

  @media (max-width: $mobile-width) {
		width: 70%;
	}

  @media (max-width: $phone-width) {
		width: 100%;
	}
}

.client-logos {
  margin: 30px 10px 0 auto;
  width: 92%;
  display: grid;
  grid-template-columns: 20.5% 20.5% 20.5% 20.5%;
  grid-column-gap: 2%;

  > div {
    height: 100px;
    align-content: center;

    position: relative;

    > img {
      max-height: 100%;
      max-width: 100%;
      vertical-align: middle;

      position: absolute;
      top: 50%;
      left: 50%;
      
      transform: translate(-50%, -50%);

      &:hover {
        cursor: pointer;
      }
    }
  }

  @media (max-width: $desktop-width) {
    width: 90%;
	}

  @media (max-width: $phone-width) {
    width: 100%;
    display: grid;
    grid-template-columns: 40% 40%;
    grid-column-gap: 20%;
	}
}