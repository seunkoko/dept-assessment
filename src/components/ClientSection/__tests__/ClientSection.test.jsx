import React from 'react';

import { shallow } from 'enzyme';

import ClientSection from '../ClientSection';

describe('ClientSection', () => {
	const component = shallow(<ClientSection />);

	it('should render ClientSection component', () => {
		expect(component).toMatchSnapshot();
  });
  
});