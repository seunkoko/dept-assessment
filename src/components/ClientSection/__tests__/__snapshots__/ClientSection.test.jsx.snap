// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientSection should render ClientSection component 1`] = `
<Fragment>
  <div
    className="client-style"
  >
    <p
      className="client-title"
    >
      CLIENTS
    </p>
    <p
      className="client-description"
    >
      We value a great working relationship with our clients above all else. It’s why they often come to our parties. It’s also why we’re able to challenge and inspire them to reach for the stars.
    </p>
    <div
      className="client-logos"
    >
      <div>
        <img
          alt="logo"
          src="test-file-stub"
        />
      </div>
      <div>
        <img
          alt="logo"
          src="test-file-stub"
        />
      </div>
      <div>
        <img
          alt="logo"
          src="test-file-stub"
        />
      </div>
      <div>
        <img
          alt="logo"
          src="test-file-stub"
        />
      </div>
      <div>
        <img
          alt="logo"
          src="test-file-stub"
        />
      </div>
      <div>
        <img
          alt="logo"
          src="test-file-stub"
        />
      </div>
      <div
        className="hide-on-phone"
      >
        <img
          alt="logo"
          src="test-file-stub"
        />
      </div>
      <div
        className="hide-on-phone"
      >
        <img
          alt="logo"
          src="test-file-stub"
        />
      </div>
      <div
        className="hide-on-phone"
      >
        <img
          alt="logo"
          src="test-file-stub"
        />
      </div>
      <div
        className="hide-on-phone"
      >
        <img
          alt="logo"
          src="test-file-stub"
        />
      </div>
      <div
        className="hide-on-phone"
      >
        <img
          alt="logo"
          src="test-file-stub"
        />
      </div>
      <div
        className="hide-on-phone"
      >
        <img
          alt="logo"
          src="test-file-stub"
        />
      </div>
      <div
        className="hide-on-phone"
      >
        <img
          alt="logo"
          src="test-file-stub"
        />
      </div>
      <div
        className="hide-on-phone"
      >
        <img
          alt="logo"
          src="test-file-stub"
        />
      </div>
      <div
        className="hide-on-phone"
      >
        <img
          alt="logo"
          src="test-file-stub"
        />
      </div>
      <div
        className="hide-on-phone"
      >
        <img
          alt="logo"
          src="test-file-stub"
        />
      </div>
    </div>
  </div>
</Fragment>
`;
